### PR TITLE
Serial Consistency QueryConfig field validation

### LINKED ISSUE #2507: Create validator to restrict consistency configuration
* for serial consistency only `SERIAL` and `LOCAL_SERIAL` are allowed
* check if needed for consistency, but guess not as `@NotNull` should be enough..

### PR DESCRIPTION
**Pull request by maheshrajamani**
**What this PR does**:
Serial Consistency config validation
**Which issue(s) this PR fixes**:
Fixes #2507 

**Checklist**
- [x] Changes manually tested
- [x] Automated Tests added/updated
- [ ] Documentation added/updated
- [x] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: apis/sgv2-quarkus-common/src/main/java/io/stargate/sgv2/api/common/config/SerialConsistencyTypeAnnotation.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import javax.validation.Constraint;
+import javax.validation.Payload;
+
+@Target({METHOD, FIELD, ANNOTATION_TYPE, CONSTRUCTOR, PARAMETER, TYPE_USE})
+@Retention(RetentionPolicy.RUNTIME)
+@Constraint(validatedBy = SerialConsistencyValidator.class)
+public @interface SerialConsistencyTypeAnnotation {
```
Comment (ivansenic) [inline apis/sgv2-quarkus-common/src/main/java/io/stargate/sgv2/api/common/config/SerialConsistencyTypeAnnotation.java:15]: maybe `SerialConsistencyValid` sounds better

File: apis/sgv2-quarkus-common/src/main/java/io/stargate/sgv2/api/common/config/SerialConsistencyValidator.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+public class SerialConsistencyValidator
+    implements ConstraintValidator<SerialConsistencyTypeAnnotation, QueryOuterClass.Consistency> {
+  private QueryOuterClass.Consistency[] subset;
+
+  @Override
+  public void initialize(SerialConsistencyTypeAnnotation constraint) {
+    this.subset = constraint.anyOf();
```
Comment (ivansenic) [inline apis/sgv2-quarkus-common/src/main/java/io/stargate/sgv2/api/common/config/SerialConsistencyValidator.java:14]: * can `anyOf` be null in any case?
* transform to list already here (or sort the array and use binary serach in the `isValid` if you wanna be fancy ;)

File: apis/sgv2-quarkus-common/src/test/java/io/stargate/sgv2/api/common/config/SerialConsistencyValidatorTest.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+import javax.validation.Validation;
+import javax.validation.Validator;
+import javax.validation.ValidatorFactory;
+import javax.validation.constraints.NotNull;
+import org.junit.jupiter.api.Test;
+
+public class SerialConsistencyValidatorTest {
+
+  private record TestRecord(
+      @NotNull
```
Comment (ivansenic) [inline apis/sgv2-quarkus-common/src/test/java/io/stargate/sgv2/api/common/config/SerialConsistencyValidatorTest.java:18]: remove not null and add test with `null`, must be valid

**Review by ivansenic — APPROVED** [review]
LGTM

File: apis/sgv2-quarkus-common/src/main/java/io/stargate/sgv2/api/common/config/SerialConsistencyTypeAnnotation.java (hunk 1 above)
Comment (maheshrajamani) [reply apis/sgv2-quarkus-common/src/main/java/io/stargate/sgv2/api/common/config/SerialConsistencyTypeAnnotation.java]: Renamed.

File: apis/sgv2-quarkus-common/src/test/java/io/stargate/sgv2/api/common/config/SerialConsistencyValidatorTest.java (hunk 3 above)
Comment (maheshrajamani) [reply apis/sgv2-quarkus-common/src/test/java/io/stargate/sgv2/api/common/config/SerialConsistencyValidatorTest.java]: @NotNull removed and null is handled in the validator.
